### PR TITLE
MAT-7184: fetch full library object in preparation to draft a library

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-7184](https://jira.cms.gov/browse/MAT-7184)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/cqlLibraryList/CqlLibraryList.tsx [hunk 1]
```diff
@@ -355,10 +355,23 @@ export default function CqlLibraryList({ cqlLibraryList, onListUpdate }) {
                 <button
                   data-testid={`create-new-draft-${selectedCQLLibrary.id}-button`}
                   onClick={() => {
-                    setCreateDraftDialog({
-                      open: true,
-                      cqlLibrary: selectedCQLLibrary,
-                    });
+                    cqlLibraryServiceApi
```
Comment (sb-prateekkeerthi) [inline src/components/cqlLibraryList/CqlLibraryList.tsx]: can we not write the logic code for onclick in separate function as we are calling the api and catching the errors

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/components/cqlLibraryList/CqlLibraryList.tsx (hunk 1 above)
Comment (nmorasb) [reply src/components/cqlLibraryList/CqlLibraryList.tsx]: just pushed a change to move the logic to its own function.
Also, this will (hopefully soon) become irrelevant as we should refactor how this works for this tech debt item: [MAT-7185](https://jira.cms.gov/browse/MAT-7185)

**Review by sb-prateekkeerthi — APPROVED** [review]
